src/transaction/log_append.cpp

```diff
@@ -1569,6 +1571,157 @@ prior_is_tde_encrypted (const log_prior_node *node)
   return node->tde_encrypted;
 }
 
+static size_t
+prior_node_serialized_size (const log_prior_node &node)
+{
+  size_t total_size = 0;
+
+  // fixed data
+  total_size += sizeof (node.log_header)
+		+ sizeof (node.start_lsa)
+		+ sizeof (node.tde_encrypted)
+		+ sizeof (node.data_header_length)
+		+ sizeof (node.ulength)
+		+ sizeof (node.rlength);
+
+  // variable data
+  total_size += static_cast<size_t> (node.data_header_length);
+  total_size += static_cast<size_t> (node.ulength);
+  total_size += static_cast<size_t> (node.rlength);
+
+  return total_size;
+}
+
+static size_t
+prior_list_serialized_size (const log_prior_node *head)
+{
+  size_t total_size = 0;
+  for (const log_prior_node *nodep = head; nodep != nullptr; nodep = nodep->next)
+    {
+      total_size += prior_node_serialized_size (*nodep);
+    }
+  return total_size;
+}
+
+static void
+prior_node_serialize (const log_prior_node &node, std::string &serialized)
+{
+#define copy_mem(mem, memsz) if ((memsz) > 0) serialized.append (mem, memsz)
```

Comment:
Nitpicking, couldn't these macros be some [templated, if needed] inline functions?